### PR TITLE
Show reserved seats.

### PR DESCRIPTION
**Pull request by Fenho**
## What?
Added reservation functionality. The room also loads the previously reserved seats.

## Why?
Main functionality.

## Tests?
I manually tested reserving one and more seats in different requests. After that I went to the room page to check if those reservations actually show up.

## Anything else?
The reservation is not associated with the schedule, so when the room loads the previous reservations, those reservations may live in different time spaces 🤯 (adding a scheduleId to reservation should fix the problem)

### CODE REVIEW

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -21,17 +22,26 @@ def edit
...
+    render :json => payload, :status => 201
+    # respond_to do |format|
+    #   if @reservation.save
+        # format.html { redirect_to room.cinema, notice: "Reservation was successfully created." }
+      #   format.json { render :show, status: :created, location: room.cinema }
+      # else
+      #   format.html { render :new, status: :unprocessable_entity }
+      #   format.json { render json: @reservation.errors, status: :unprocessable_entity }
+      # end
+    # end
```
Comment (dasolari) [inline app/controllers/reservations_controller.rb]: Delete this.

File: app/javascript/packs/room.jsx [hunk 2]
```diff
@@ -5,11 +5,33 @@ import { rows, columns, seats, initialSeatState } from '../constants';
...
+      const numberIndex = columns.indexOf(number);
+      setSeatStatus((prevState) => {
+        const stateCopy = prevState.map((row) => row.slice());
+        stateCopy[letterIndex][numberIndex] = !selectedSeats.includes(seat);
+        return stateCopy;
+      });
+    })
+  }
+
+  const selectSeat = (seat, letter, number) => {
```
Comment (dasolari) [inline app/javascript/packs/room.jsx:34]: Nice, lo cambiaste a `seat`.

File: app/views/rooms/show.html.erb [hunk 3]
```diff
@@ -1,3 +1,8 @@
 <!-- <%= link_to 'Back', @movie, class: "back-btn" %> -->
 
-<%= javascript_pack_tag 'room' %>
\ No newline at end of file
+<%= javascript_pack_tag 'room' %>
+<%= content_tag :div,
+  id: "room_data",
+  reservations: @reservations.to_json,
+  data: @room.to_json do %>
```
Comment (dasolari) [inline app/views/rooms/show.html.erb:7]: Este `do` está bien?

Comment (Fenho) [reply app/views/rooms/show.html.erb:7]: Asi es como encontre que se pasaban props, no se lo que hace

Comment (Fenho) [reply app/views/rooms/show.html.erb:7]: Confirmo que sin eso no funciona. Si no se pone ese do te entrega toda la info como un json en vez de html.

**Review by dasolari — APPROVED** [review]
Thanks for the fixes!

**Pull request merged**